src/JestExt.ts

```diff
@@ -364,6 +369,8 @@ export class JestExt {
     const statusList = this.testResultProvider.updateTestResults(normalizedData)
     updateDiagnostics(statusList, this.failDiagnostics)
 
+    this.sidebarProvider.refresh(data)

```

Comment:
do we need to process all the tests up front like this, giving we will only need to display the tree when the file is actually displayed in the editor... please see if you can move the logic to `triggerUpdateActiveEditor` instead.

Reply:
indeed, I was thinking about the outline view, my bad.

trying it again on vscode-jest itself, added a few fake tests and encountered the following issues:
- looks like under the root "Tests" are the describe blocks instead of the file.  So if a test file doesn't have a root describe block or it has multiple parallel describe blocks, they will not show any hierarchical relationship. Is that intended? Is there a particular reason we don't want to use the test file for grouping?
- nested describe blocks are not shown.
- the root level `it` blocks are not shown, and if reload/restart vscode, it crashed the plugin, put it in the never-ending spinning "Starting watch mode" and the Test sidebar is empty... (same error as mentioned in earlier comment.)

Reply:
I added support for root tests, removed the folder name from filename and changed the default setting to show files by default. My practice has always been to have all tests inside describe and to have only one describe per file. That's why I chose the implementation I did.
I see you have to reload the window when the settings change but I think that's the case for all vscode-jest settings because `getExtensionSettings` is called on `extension.activate` event.